modules/dnn/src/darknet/darknet_io.cpp

```diff
@@ -0,0 +1,638 @@
+//
+//THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+//SOFTWARE.
+//
+//M*/
+
+#include <opencv2/core.hpp>
+
+#include <iostream>
+#include <fstream>
+#include <sstream>
+
+#include "darknet_io.hpp"
+
+namespace darknet {
+
+    template<typename T>
+    std::string toString(const T &v)
+    {
+        std::ostringstream ss;
+        ss << v;
+        return ss.str();
+    }
+
+    template<typename T>
+    T getParam(const std::map<std::string, std::string> &params, const std::string param_name, T init_val)
+    {
+        std::map<std::string, std::string>::const_iterator it = params.find(param_name);
+        if (it != params.end()) {
+            std::stringstream ss(it->second);
+            ss >> init_val;
+        }
+        return init_val;
+    }
+
+    class setLayersParams {
+
+        NetParameter *net;
+        int layer_id;
+        std::string last_layer;
+        std::vector<std::string> fused_layer_names;
+
+    public:
+        setLayersParams(NetParameter *_net, std::string _first_layer = "data") :
+            net(_net), layer_id(0), last_layer(_first_layer)
+        {}
+
+        void setLayerBlobs(int i, std::vector<cv::Mat> blobs)
+        {
+            cv::dnn::experimental_dnn_v1::LayerParams &params = net->layers[i].layerParams;
+            params.blobs = blobs;
+        }
+
+        void setLastLayerName(std::string layer_name)
+        {
+            if(net->layers.size() > 0)
+                net->layers.rbegin()->layer_name = layer_name;
+        }
+
+        cv::dnn::experimental_dnn_v1::LayerParams getParamConvolution(int kernel, int pad,
+            int stride, int filters_num)
+        {
+            cv::dnn::experimental_dnn_v1::LayerParams params;
+            params.name = "Convolution-name";
+            params.type = "Convolution";
+
+            params.set<int>("kernel_size", kernel);
+            params.set<int>("pad", pad);
+            params.set<int>("stride", stride);
+
+            params.set<bool>("bias_term", false);	// true only if(BatchNorm == false)
+            params.set<int>("num_output", filters_num);
+
+            return params;
+        }
+
+
+        void setConvolution(int kernel, int pad, int stride,
+            int filters_num, int channels_num, int use_batch_normalize, int use_relu)
+        {
+            cv::dnn::experimental_dnn_v1::LayerParams conv_param =
+                getParamConvolution(kernel, pad, stride, filters_num);
+
+            darknet::LayerParameter lp;
+            std::string layer_name = "conv_" + toString(layer_id);
```

Comment:
Please try to use `cv::format("conv_%d", layer_id)` instead `toString` here and in other places.